pkg/app/launcher/cmd/launcher/launcher.go

```diff
@@ -267,6 +305,21 @@ func (l *launcher) run(ctx context.Context, input cli.Input) error {
 		return nil
 	}
 
+	commandHandler := func(ctx context.Context, cmdCh <-chan model.ReportableCommand) {
+		input.Logger.Info("started a worker for handling restart piped command")
+		for {
+			select {
+			case cmd := <-cmdCh:
+				l.handleCommand(ctx, input, cmd)
+
+			case <-ctx.Done():
+				input.Logger.Info("a worker has been stopped")
+				return
+			}
+		}
+	}
+	go commandHandler(ctx, l.commandCh)
```

Comment:
How about using goroutine with errorGroup instead of creating a simple goroutine?